storage/oasis/client.go

```diff
@@ -66,7 +66,7 @@ func (cf *ClientFactory) Consensus() (*ConsensusClient, error) {
 		if err != nil {
 			return nil, err
 		}
-		nodeApi = cobalt.NewCobaltConsensusApiLite(grpcConn)
+		nodeApi = cobalt.NewCobaltConsensusApiLite(grpcConn, (*cf.connection).Consensus())
```

Comment:
can't proceed with this. making an sdk connection to an old node might break with some combinations. e.g. early nodes didn't have a dedicated GetChainContext, and you'd instead get the whole genesis document and hash it if you really needed it

Reply:
It's not SDK, it's oasis-core. Just makes the conversions a little simpler; for example, if the struct remains the same across Cobalt and Damask but contains an enum, we can either a) interpret the result's CBOR as a Damask object, or b) interpret the result's CBOR as a Cobalt object, then manually convert every single field just so we can explicitly cast the enum.

It did work in practice, and it was only used for a select few methods. ("	// Used as a convenience for calling methods that are ABI-compatible between Cobalt and Damask.", said the docstring for `damaskClient` in the cobalt client).

I don't think it was risky, because Cobalt is frozen in time, so if it worked once, it would work always. That said, it was hacky. Removed in favor of explicit conversions.

Reply:
I meant cf.connection was connection.Connection from `connection "github.com/oasisprotocol/oasis-sdk/client-sdk/go/connection"`, from the SDK, i.e. tracks latest rather than being pinned to Damask. Thanks for the changes.